copy/copy.go

```diff
@@ -489,6 +501,18 @@ func (ic *imageCopier) copyBlobFromStream(srcStream io.Reader, srcInfo types.Blo
 		inputInfo.Size = -1
 	}
 
+	// the pipe is here to facilitate the stream copier which will act as an
+	// intermediary to signal progress information to any registered channel.
+	if ic.progress != nil && ic.progressInterval > 0 {
```

Comment:
The comment about the pipe logically seems to belong inside the `if`?

Above the `if`, please add something like
```go
// === Report progress using the ic.progress channel, if required
```
(i.e. saying _what_ this part of the pipeline does, before talking about _how_)

BTW, is it intentional that this is after the compression unlike the `bar.NewProxyReader` progress reporting which is a bit above?